dash-renderer/src/actions/index.js

```diff
@@ -525,6 +525,33 @@ const getVals = input =>
 
 const zipIfArray = (a, b) => (Array.isArray(a) ? zip(a, b) : [[a, b]]);
 
+function inputsToDict(inputs_list) {
+    // Ported directly from _utils.py, inputs_to_dict
+    // takes an array of inputs (some inputs may be an array)
+    // returns an Object (map):
+    //  keys of the form `id.property` or `{"id": 0}.property`
+    //  values contain the property value
+    if (!inputs_list) {
+        return {};
+    }
+    const inputs = {};
+    for (let i = 0; i < inputs_list.length; i++) {
+        if (Array.isArray(inputs_list[i])) {
+            const inputsi = inputs_list[i];
+            for (let ii = 0; ii < inputsi.length; ii++) {
+                const id_str = `${JSON.stringify(inputsi[ii].id)}.${
+                    inputsi[ii].property
+                }`;
+                inputs[id_str] = inputsi[ii].value ? inputsi[ii].value : null;
+            }
+        } else {
+            const id_str = `${inputs_list[i].id}.${inputs_list[i].property}`;
```

Comment:
This can get object IDs too, with the `MATCH` wildcard.

For both this and the array case above lets use the [`stringifyId` function](https://github.com/plotly/dash/blob/dev/dash-renderer/src/actions/dependencies.js#L118) which handles both strings and objects, and ensures sorted object keys.

Reply:
Oh cool. Is this what you mean? This should still work for wildcard object IDs.
``` JavaScript
const id_str = `${stringifyId(inputs_list[i].id)}.${inputs_list[i].property}`;
```
(same for the array case)